pkg/postgres/repository.go

```diff
@@ -399,6 +416,58 @@ ORDER BY estimates.identifier ASC`,
 	return
 }
 
+//GetRecentlyActiveRuns collects all the data about any runs that have been updated
+//since touchThreshold. The Run#Finished method can be used to determine which runs
+//have arrived at their terminal station, and can therefore be removed from state.
+func (a *RepositoryAgent) GetLatestEstimates(station martaapi.Station) (res []LastestEstimate, err error) {
+	rows, err := a.DB.Query(`
+WITH station_estimates AS (
+  SELECT runs.run_group_identifier,
+    estimates.estimated_arrival_time,
+    estimates.estimate_moment,
+    ROW_NUMBER() OVER (PARTITION BY runs.run_group_identifier
+      ORDER BY estimates.estimate_moment DESC) AS rank
```

Comment:
Group the estimates by arrival and then rank them (within their groups) by recency so that (below) we can fetch the latest from each group. I should probably add an index to help with this query, but in the future I'd like to just track the latest estimate directly on the `arrivals` table and move away from the `estimates` table altogether, so I think I'm going to not add the index